feat: implement warnings on NeverClones that implement Copy

donmai-me opened the pull request:
implements #393

LukeMathWalker commented:
The implementation looks good!
Can you add a UI test for it?

You can use [this one](https://github.com/LukeMathWalker/pavex/tree/main/libs/ui_tests/blueprint/constructors/a_warning_is_emitted_for_unused_constructors) as a reference.
The test should live [in this folder](https://github.com/LukeMathWalker/pavex/tree/main/libs/ui_tests/blueprint/common).

donmai-me commented:
Are ui tests generated by a pavex command? I noticed that the name in `a_warning_is_emitted_for_unused_constructors`'s cargo.toml is
```toml
[package]
name = "app_30248169"
```

Sorry if I missed a piece of documentation somewhere

LukeMathWalker commented:
Not yet, although I should create a helper CLI to bootstrap them!
That's the hash of the crate. The way I usually go about it—name it "app_<random sequence of chars>" and when you run `cargo test` it'll complain about missing a specific `app_XXXXX`, then I rename it to that 😅 Not the most polished, I know.

donmai-me commented:
Sorry for the long delay. I redid the pr and I think I implemented the ui test. I just renamed what pavexc wrote to expectations directory and the test passes on my machine. Hope I'm doing things right

LukeMathWalker commented:
/ok-to-test sha=ddc9c04
